### PR TITLE
ndvi review

### PR DESCRIPTION
**Pull request by mred1208**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [x] Are the data downloaded in the code
- [x] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [x] Do functions follow PEP 8 format conventions?
- [x] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [x] Are function names expressive (the name describes what the function does)?
- [x] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [x] Are variable names throughout the code, expressive?
- [x] Suggest changes if not, highlight what was done well
- [x] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [x] Are segments of code repeated in the file or is the code DRY?
- [x] Are loops used to optimize DRY code?
- [x] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

File: ea-2021-04-ndvi-automation.py [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+# with data cleaned to deal with clouds
+
+# YOUR CODE HERE
+f, ax = plt.subplots(figsize=(10,10))
+
+
+for site, data in all_ndvi_mean_df_date_index.groupby('site'):
+    ax.scatter(data.index, data.mean_ndvi, label=site)
+    
+ax.set(xlabel='Date')
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:605]: For time-series with the date series on the x axis, line or bar charts tend to read better than scatter plots

File: ea-2021-04-ndvi-automation.py [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+harv_landsat_dirs = sorted(glob(os.path.join(harv_landsat_dir, "LC08*")))
+
+
+
+sjer_landsat_dir = os.path.join(sites[0],
+                           "landsat-crop")
+sjer_landsat_dirs = sorted(glob(os.path.join(sjer_landsat_dir, "LC08*")))
+
+
+landsat_dirs_all = harv_landsat_dirs + sjer_landsat_dirs
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:465]: can be good to put to put this kind of thing into a function, but only really necessary if there were more than 2 scenes and you want to trim the code

File: ea-2021-04-ndvi-automation.py [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+
+
+all_bands_harv = []
+for aband in band_paths:
+    print("Opening up", aband)
+    cleaned_band = open_clean_bands(band_path=aband, valid_range=(0, 10000))
+    all_bands_harv.append(cleaned_band)
+    date = aband[50:58]
+    
+# Open up the cloud mask layer
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:313]: duplicate of below (row 319)?

File: ea-2021-04-ndvi-automation.py [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+ndvi_clean = mask_crop_ndvi(all_bands=all_bands_harv,
+                            crop_bound=crop_bound,
+                            pixel_qa=cl_mask,
+                            vals=vals)
+
+
+mean_ndvi = np.mean(ndvi_clean)
+
+
+harv_date = pd.to_datetime(date, format='%Y%m%d')
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:354]: lines 320-360 code looks good, finding the notes a little hard to follow though (that may be just me!)

File: ea-2021-04-ndvi-automation.py [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+
+all_ndvi_mean_df_date_index = all_ndvi_mean_df.set_index('date')
+
+
+#convert mean_ndvi to floats
+all_ndvi_mean_df_date_index["mean_ndvi"] = all_ndvi_mean_df_date_index["mean_ndvi"].astype(float)
+
+all_ndvi_mean_df_date_index.sort_index(inplace=True)
+
+all_ndvi_mean_df_date_index
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:532]: good loop here, perhaps some more descriptive notes for the steps? 

vector names are expressive and easy to follow

File: ea-2021-04-ndvi-automation.py [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+    #CLOUD MASK
+    
+    qa_xr_path_each = glob(os.path.join(adir, "*qa*"))
+    # Clip the cloud mask layer
+    cl_mask = rxr.open_rasterio(qa_xr_path_each[0], masked=True).squeeze()
+    # Get a list of each directory
+    path_each = os.path.join("ndvi-automation", "sites")
+    # Get a list of both site directories
+    sites = glob(path + "/*/")
+    # Open up the shapefile for clipping your landsat data to the study area
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:488]: sites as defined here is already defined earlier in the code

File: ea-2021-04-ndvi-automation.py [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+# This is optional - if you export a **.csv** file with the columns specified above: Site, Date and NDVI Value you can get an additional 10 points.
+# 
+# * FULL CREDIT: File exists in csv format and contains the columns specified.
+# We will check your github repo for this file!
+# 
+
+# In[14]:
+
+
+all_ndvi_mean_df_date_index.to_csv(path_or_buf='/home/jovyan/ndvi_mean_df.csv')
```
Comment (mike-petrut) [inline ea-2021-04-ndvi-automation.py:676]: nice work getting the bonus points

**Review by mike-petrut — COMMENTED** [review]
@mred1208 great work on this assignment. Comments attached

File: ea-2021-04-ndvi-automation.py (hunk 5 above)
Comment (mred1208) [reply ea-2021-04-ndvi-automation.py:532]: I agree about including more descriptive notes.

File: ea-2021-04-ndvi-automation.py (hunk 1 above)
Comment (mred1208) [reply ea-2021-04-ndvi-automation.py:605]: I found it very strange looking when I originally plotted as a line graph.
